test/run.d

```diff
@@ -20,12 +20,13 @@ import tools.paths;
 
 const scriptDir = __FILE_FULL_PATH__.dirName.buildNormalizedPath;
 auto testPath(R)(R path) { return buildNormalizedPath(scriptDir, path); }
-string resultsDir = testPath("test_results");
+shared string resultsDir = testPath("test_results");
```

Comment:
I don't see how changing this to `shared` helps with anything, unless you also use atomic primitives from `core.atomic`.

Reply:
The problem is that a TLS variable is only changed by the startup code in the main thread, every other thread is unmodified.

Reply:
Ah, yes of course. `immutable` should also work.